docs/builtins.md

```diff
@@ -0,0 +1,806 @@
+
+The `print` function prints its arguments to the output area, each
+separated by a single space and terminated by a newline character. If
+no arguments are provided it only prints the newline character. 
+
+A tab is printed for the `\t` control character. A newline is printed
+for the `\n` control character. Quotes in string literals are escaped
+with backslash `\`, for example `print "Here's a tab: \t.\nShe said: \"Thank you!\""`.
+
+In a browser environment `print` typically prints to a DOM Element wired
+to the evy runtime. When running evy from the command line interface,
+`print` prints to standard out.
+
+### `read`
+ 
+`read` reads a line of input from the user and returns it as a
+string. The newline character is not included in the returned string.
+
+#### Example
+```evy
+name := read
+print "Hello, "+name+"!"
+```
+
+Input
+```
+Mary Jackson
+```
+
+Output
+```
+Hello, Mary Jackson!
+```
+
+#### Reference 
+
+    read:string
+
+The `read` function returns a string that contains the line of input
+that the user entered up until, excluding the newline character. It is
+a blocking functions, which means that it will not return until the
+user has entered a line of input and pressed the Enter key.
+
+In a browser environment `read` typically reads from a DOM Textarea
+Element wired to the evy runtime. When running evy from the command
+line interface, `read` reads from standard in.
+
+### `cls`
+
+`cls` clears the output area of all printed text.
+
+#### Example
+```evy
+print "Hello"
+sleep 1
+cls
+print "Bye"
+```
+
+Output
+```
+Bye
+```
+
+#### Reference 
+
+    cls
+
+The `cls` function clears the output area. In a browser environment
+`cls` typically clears the DOM Element wired to the evy runtime which
+is used for printing. When running evy from the command line interface,
+`cls` clears the terminal, similar to the Unix `clear` or Windows `cls`
+commands.
+
+### `printf`
+
+`printf` stands for print formatted.
+
+`printf` prints its arguments to the output area according to a _format_
+string. The format string is the first argument, and it
+contains _specifiers_. Specifiers start with a percent sign `%`. They
+tell the `printf` function how and where to print the remaining
+arguments inside the format string. The rest of the format string is
+printed to the output area without changes. 
+
+Here are some valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| `%v`      | the argument in its default format  |
+| `%q`      | a double-quoted string |
+| `%%`      | a percent sign `%`  |
+
+#### Example
+
+```evy
+printf "The tank is %%100 full.\n\n"
+
+weather := "rainy"
+printf "It is %v today.\n" weather
+rainfall := 10
+printf "There will be %vmm of rainfall.\n" rainfall
+unicorns := false
+printf "There will be unicorns eating lollipops: %v.\n\n" unicorns
+
+printf "They said: %q\n" "wow"
+printf "Array: %v\n" [1 2 3]
+printf "Map: %v\n" {a:1 b:2}
+```
+
+Output
+```
+%v string: abc
```

Comment:
This output needs to be updated for the examples above.

Reply:
oops, sorry! done.